browser/brave_vpn/test/android/vpn_receipt_browsertest.cc

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/base64.h"
+#include "base/json/json_reader.h"
+#include "base/run_loop.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/components/brave_vpn/features.h"
+#include "brave/components/brave_vpn/pref_names.h"
+#include "brave/components/skus/common/features.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/test/base/android/android_browser_test.h"
+#include "chrome/test/base/chrome_test_utils.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/url_loader_interceptor.h"
+#include "url/gurl.h"
+
+namespace brave_vpn {
+
+namespace {
+std::string GetHeadersForURL(const std::string& url) {
+  if (base::EndsWith(url, "redirect=true", base::CompareCase::SENSITIVE)) {
+    return "HTTP/1.1 302 Found\nContent-type: text/html\nLocation: "
+           "https://account.brave.com/\n\n";
+  }
+  return "HTTP/1.1 200 OK\nContent-type: text/html\n\n";
+}
+
+bool URLLoaderInterceptorCallback(
+    content::URLLoaderInterceptor::RequestParams* params) {
+  LOG(ERROR) << "spec:" << params->url_request.url.spec();
+  content::URLLoaderInterceptor::WriteResponse(
+      GetHeadersForURL(params->url_request.url.query()),
+      "<html><body></body></html>", params->client.get());
+  return true;
+}
+}  // namespace
+
+class VpnReceiptBrowserTest : public PlatformBrowserTest {
```

Comment:
why is this in android? Also why are the tests disabled?

Reply:
Test removed for now in latest commit. Disabled because it had errors and it's difficult to troubleshoot ☹️

Reply:
BTW - I'm planning on filing a follow up issue to revert the commit that removes the tests and fix them